Functions/Mock.ps1

```diff
@@ -248,104 +244,12 @@ about_Mocking
-                    $comma = if ($metadataSafeForDynamicParams.Parameters.Count -gt 0) {
-                        ','
-                    }
-                    else {
-                        ''
-                    }
-                    $dynamicParamBlock = "dynamicparam { Get-MockDynamicParameter -ModuleName '$ModuleName' -FunctionName '$CommandName' -Parameters `$PSBoundParameters -Cmdlet `$PSCmdlet }"
-
-                    $code = @"
-                        $cmdletBinding
-                        param(
-                            [object] `${P S Cmdlet}$comma
-                            $paramBlockSafeForDynamicParams
-                        )
-
-                        `$PSCmdlet = `${P S Cmdlet}
-
-                        $dynamicParamStatements
-"@
-
-                    $dynamicParamScriptBlock = [scriptblock]::Create($code)
-
-                    $sessionStateInternal = Get-ScriptBlockScope -ScriptBlock $contextInfo.Command.ScriptBlock
-
-                    if ($null -ne $sessionStateInternal) {
-                        Set-ScriptBlockScope -ScriptBlock $dynamicParamScriptBlock -SessionStateInternal $sessionStateInternal
-                    }
-                }
-            }
         }
 
-        $EscapeSingleQuotedStringContent =
-        if ($global:PSVersionTable.PSVersion.Major -ge 5) {
-            { [System.Management.Automation.Language.CodeGeneration]::EscapeSingleQuotedStringContent($args[0]) }
-        }
-        else {
-            { $args[0] -replace "['‘’‚‛]", '$&$&' }
-        }
-
-        $newContent = & $SafeCommands['Get-Content'] function:\MockPrototype
-        $newContent = $newContent -replace '#FUNCTIONNAME#', (& $EscapeSingleQuotedStringContent $CommandName)
-        $newContent = $newContent -replace '#MODULENAME#', (& $EscapeSingleQuotedStringContent $ModuleName)
-
-        $canCaptureArgs = 'true'
-        if ($contextInfo.Command.CommandType -eq 'Cmdlet' -or
-            ($contextInfo.Command.CommandType -eq 'Function' -and $contextInfo.Command.CmdletBinding)) {
-            $canCaptureArgs = 'false'
-        }
-        $newContent = $newContent -replace '#CANCAPTUREARGS#', $canCaptureArgs
-
-        $code = @"
-            $cmdletBinding
-            param ( $paramBlock )
-            $dynamicParamBlock
-            begin
-            {
-                `${mock call state} = @{}
-                $($newContent -replace '#BLOCK#', 'Begin' -replace '#INPUT#')
-            }
-
-            process
-            {
-                $($newContent -replace '#BLOCK#', 'Process' -replace '#INPUT#', '-InputObject @($input)')
-            }
-
-            end
-            {
-                $($newContent -replace '#BLOCK#', 'End' -replace '#INPUT#')
-            }
-"@
-
-        $mockScript = [scriptblock]::Create($code)
+        $mockBlocks = New-CodeBlocksForMock -ContextInfo $contextInfo -Metadata $metadata
```

Comment:
The generation of the mock function body was moved to the new `New-CodeBlocksForMock` function.

It helps to have a cleaner and smaller `Mock` function by delegating the scriptblocks generation to the new function